gradoop-store/gradoop-hbase/src/main/java/org/gradoop/flink/io/impl/hbase/inputformats/EdgeTableInputFormat.java

```diff
@@ -78,114 +58,39 @@ public EdgeTableInputFormat(EdgeHandler<E, V> edgeHandler,
   }
 
   /**
-   * {@inheritDoc}
+   * Get the scanner instance. If a query was applied to the elementHandler,
+   * the Scan will be extended with a HBase filter representation of that query.
+   *
+   * @return the Scan instance with an optional HBase filter applied
    */
   @Override
   protected Scan getScanner() {
     Scan scan = new Scan();
     scan.setCaching(HBaseConstants.HBASE_DEFAULT_SCAN_CACHE_SIZE);
 
-    FilterList conjunctFilters = new FilterList(FilterList.Operator.MUST_PASS_ALL);
+    if (edgeHandler.getQuery() != null) {
```

Comment:
Maybe its possible to introduce a additional ElementTableInputFormat to reduce duplicated code (VertexTableInputFormat, GraphHeadTableInputFormat, EdgeTableInputFormat)

Reply:
Because of generics definition it is not possible. To do that you have to define the `.getQuery()` function in a higher interface like `ElementHandler`. But ElementHandler doesn't have any generics information.